irc.py

```diff
@@ -20,15 +19,20 @@ def change_nick(self, nick):
         self.irc.send("NICK " + nick + "\n")
 
     def connect(self, server, channel, botnick):
-        #defines the socket
+        # Defines the socket
         print("connecting to:"+server)
-        self.irc.connect((server, 6667))                                                         #connects to the server
-        self.irc.send("USER " + botnick + " " + botnick +" " + botnick + " :This is a fun bot!\n") #user authentication
+        # Connects to the server
+        self.irc.connect((server, 6667))
+        # User authentication
+        self.irc.send("USER " + botnick + " " + botnick + " " +
+                      botnick + " :This is a fun bot!\n")
```

Comment:
This could be simplified by rewriting the line as `self.irc.send("USER " + (botnick + " ")*3 + ":This is a fun bot!\n")`. Could be worth checking out

Reply:
To begin with,
>> self.irc.send("USER " + (botnick + " ")*3 + ":This is a fun bot!\n")  //recommended
has 
>> botnick + " " + botnick + " " + botnick + " "
which is not the expected input as compared to botnick + " " + botnick + " " + botnick    //original
I gave it a try with 
>>  self.irc.send("USER " + (botnick + " ")*2 + botnick + ":This is a fun bot!\n")  

and this is what I got:
>> Traceback (most recent call last):
  File "spider_bot.py", line 57, in <module>
    spider_bot(irc, spider_bot)
  File "spider_bot.py", line 36, in spider_bot
    irc.connect(server, channel, nickname)
  File "/home/nakulkar/irc_bots/irc.py", line 27, in connect
    self.irc.send("USER " + (botnick + " ")*2 + botnick +
TypeError: unsupported operand type(s) for +: 'function' and 'str'

Reply:
To begin with,
>> self.irc.send("USER " + (botnick + " ")*3 + ":This is a fun bot!\n")  //recommended
has 
>> botnick + " " + botnick + " " + botnick + " "
which is not the expected input as compared to botnick + " " + botnick + " " + botnick    //original
I gave it a try with 
>>  self.irc.send("USER " + (botnick + " ")*2 + botnick + ":This is a fun bot!\n")  
and it works!